sktime/forecasting/arima.py

```diff
@@ -266,26 +294,52 @@ def __init__(self, start_p=2, d=None, start_q=2, max_p=5,
         super(AutoARIMA, self).__init__()
 
         # import inside method to avoid hard dependency
-        from pmdarima.arima import AutoARIMA as _AutoARIMA
+        try:
```

Comment:
I'd move this to the top of the file, not inside the class, to make it easier to see with the other imports. I'd also write a helper function to do this which we can use in multiple places. I've updated #325 to give an example! 

Checking for importing the main package should also be enough instead of trying all submodules, no? Making sure people install the correct version may be another issue. That's where EXTRA_REQUIRES may come in handy ...

Reply:
Yes, I agree with this. I did see your suggestion of performing a check at the top of files, but didn't realise there'd be so many dependencies scattered throughout sorry. I'll change them to function calls at the top of files.